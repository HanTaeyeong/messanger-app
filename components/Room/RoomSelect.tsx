import React from 'react'

const mockRooms = [
  { id: '12', name: 'first room' },
  { id: '53', name: 'second room' }
]

function RoomSelect() {

  return (
    <div className='room-list'>
      {mockRooms.map(room =>
        <div className='room-list-item' key={room.id} >{room.name}</div>
      )}
    </div>
  )
}

export default RoomSelect