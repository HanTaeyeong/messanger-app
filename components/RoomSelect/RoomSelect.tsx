import React from 'react'
import Link from 'next/link'


const mockRooms = [
  { id: '12', name: 'first room' },
  { id: '53', name: 'second room' }
]

function RoomSelect() {

  return (
    <div className='room-list'>
      <div>Create Room</div>

      {mockRooms.map(room =>
        <div key={room.id}>
          <Link href='/room/[roomId]' as={`/room/${room.id}`} >
            <a className='room-list-item' id={room.id} >
              {room.name}
            </a>
          </Link>
        </div>
      )}

      <div>Search Room</div>

    </div>
  )
}

export default RoomSelect