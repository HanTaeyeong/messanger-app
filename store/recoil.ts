import { atom, selector, useRecoilState } from 'recoil';

export const weekList: String[] = ['Sunday', 'Monday', 'Tuesday', 'Wendsday', 'Thursday', 'Fryday', 'Saturday']

export const testDate = atom<Number>({
    key: 'testDate',
    default: 0,
});

export const testWeekName = selector<String>({
    key: 'testWeekName',
    get: ({ get }) => weekList[(+get(testDate) % 7) | 0] || 'Sunday',
    set: ({ set }) => { }
});
