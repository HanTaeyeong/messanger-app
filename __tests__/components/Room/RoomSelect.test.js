import React from 'react'

import TestRenderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil'

import RoomSelect from '../../../components/Room/RoomSelect'


describe('Login Component test', () => {
    let testComponent;

    beforeEach(() => {
        testComponent = TestRenderer.create(
            <RecoilRoot>
                <RoomSelect />
            </RecoilRoot>)
    })

    test('renders well', () => {
        expect(testComponent).toBeTruthy()
        expect(testComponent.toJSON()).toMatchSnapshot();
    })

    test('room list item test', () => {
        expect(testComponent.root.findByProps({ className: "room-list" })
            .children[0].children[0]).toBe('first room')
    })

})
