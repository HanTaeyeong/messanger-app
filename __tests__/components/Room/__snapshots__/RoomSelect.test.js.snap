// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component test renders well 1`] = `
<div
  className="room-list"
>
  <div
    className="room-list-item"
  >
    first room
  </div>
  <div
    className="room-list-item"
  >
    second room
  </div>
</div>
`;
